tico/serialize/operators/op_avg_pool2d.py

```diff
@@ -57,7 +57,12 @@ def has_padding(self, args: AvgPool2dArgs) -> bool:
             return True
 
     def has_same_padding(self, args: AvgPool2dArgs) -> bool:
-        input_shape = list(extract_shape(args.input))
+        input_shape, input_shape_signature = extract_circle_shape(args.input)
+
+        if input_shape_signature is not None:
+            # TODO: support dynamic shapes
+            raise NotImplementedError("Dynamic shape is not supported yet")
```

Comment:
Are these guards added because there's no test modules? Or, they really can't be run?

Reply:
Or if it's the functionality of this, then I think it should be renamed.

Reply:
Just for note, blocking dynamic dimensions only when the batch dimension is not dynamic could be a good idea as well, since in most cases it's the batch that's dynamic. However, blocking all dynamic dimensions entirely also makes sense, as it clearly defines the scope of coverage.